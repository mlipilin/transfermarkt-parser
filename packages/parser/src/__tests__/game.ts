import { game } from '../index'

const timeout = Number(process.env.JEST_ASYNC_TIMEOUT)

describe('list', () => {
  test(
    'game.list("GB1", "2022") -> list of games for EPL (season 22/23, matchday 1)',
    async () => {
      await expect(game.list('GB1', '2022', 1)).resolves.toMatchSnapshot()
    },
    timeout
  )
  test(
    'game.list("wrongCompetitionId", "wrongSeasonId", -999) -> []',
    async () => {
      await expect(
        game.list('wrongCompetitionId', 'wrongSeasonId', -999)
      ).resolves.toMatchSnapshot()
    },
    timeout
  )
  test(
    'game.list() -> []',
    async () => {
      // @ts-ignore
      await expect(game.list()).resolves.toMatchSnapshot()
    },
    timeout
  )
})
