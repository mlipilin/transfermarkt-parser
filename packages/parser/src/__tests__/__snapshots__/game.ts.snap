// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list game.list("GB1", "2022") -> list of games for EPL (season 22/23, matchday 1) 1`] = `
[
  {
    "club1Goals": 0,
    "club1Id": 873,
    "club2Goals": 2,
    "club2Id": 11,
    "competitionId": "GB1",
    "id": "3837814",
    "matchday": 1,
    "seasonId": "2022",
  },
  {
    "club1Goals": 2,
    "club1Id": 931,
    "club2Goals": 2,
    "club2Id": 31,
    "competitionId": "GB1",
    "id": "3837815",
    "matchday": 1,
    "seasonId": "2022",
  },
  {
    "club1Goals": 4,
    "club1Id": 148,
    "club2Goals": 1,
    "club2Id": 180,
    "competitionId": "GB1",
    "id": "3837816",
    "matchday": 1,
    "seasonId": "2022",
  },
  {
    "club1Goals": 2,
    "club1Id": 762,
    "club2Goals": 0,
    "club2Id": 703,
    "competitionId": "GB1",
    "id": "3837817",
    "matchday": 1,
    "seasonId": "2022",
  },
  {
    "club1Goals": 2,
    "club1Id": 399,
    "club2Goals": 1,
    "club2Id": 543,
    "competitionId": "GB1",
    "id": "3837819",
    "matchday": 1,
    "seasonId": "2022",
  },
  {
    "club1Goals": 2,
    "club1Id": 989,
    "club2Goals": 0,
    "club2Id": 405,
    "competitionId": "GB1",
    "id": "3837820",
    "matchday": 1,
    "seasonId": "2022",
  },
  {
    "club1Goals": 0,
    "club1Id": 29,
    "club2Goals": 1,
    "club2Id": 631,
    "competitionId": "GB1",
    "id": "3837821",
    "matchday": 1,
    "seasonId": "2022",
  },
  {
    "club1Goals": 2,
    "club1Id": 1003,
    "club2Goals": 2,
    "club2Id": 1148,
    "competitionId": "GB1",
    "id": "3837818",
    "matchday": 1,
    "seasonId": "2022",
  },
  {
    "club1Goals": 1,
    "club1Id": 985,
    "club2Goals": 2,
    "club2Id": 1237,
    "competitionId": "GB1",
    "id": "3837822",
    "matchday": 1,
    "seasonId": "2022",
  },
  {
    "club1Goals": 0,
    "club1Id": 379,
    "club2Goals": 2,
    "club2Id": 281,
    "competitionId": "GB1",
    "id": "3837823",
    "matchday": 1,
    "seasonId": "2022",
  },
]
`;

exports[`list game.list("wrongCompetitionId", "wrongSeasonId", -999) -> [] 1`] = `[]`;

exports[`list game.list() -> [] 1`] = `[]`;
